### PR TITLE
[RPC] Replace object lookup with event

### PR DESCRIPTION
**Pull request by damirka**
## Description 

Display may be created at any time, wrapped and edited without visible changes unless `update_display` is called to explicitly bump the version. Additionally, we have the base for version support.

## Test Plan 

Existing tests should pass.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

Display listens to the `display::VersionUpdated` event instead of the object lookup.

### CODE REVIEW

File: crates/sui-json-rpc/src/read_api.rs [hunk 1]
```diff
@@ -927,62 +926,30 @@ async fn get_display_fields(
...
-            .data
-            .try_as_move()
-            .ok_or_else(|| anyhow!("Failed to extract Move object from {display_object_id}"))?;
-        Ok(Some(
-            bcs::from_bytes::<DisplayObject>(move_object.contents()).map_err(|e| {
-                anyhow!("Failed to deserialize DisplayObject {display_object_id}: {e}")
-            })?,
-        ))
+        .query_events(
+            EventFilter::MoveEventType(DisplayVersionUpdatedEvent::type_(object_type)),
```
Comment (666lcz) [inline crates/sui-json-rpc/src/read_api.rs:934]: Does the creation of display object also trigger`DisplayVersionUpdatedEvent`?

File: crates/sui-json-rpc/src/read_api.rs [hunk 2]
```diff
@@ -927,62 +926,30 @@ async fn get_display_fields(
...
+    // If there's any recent version of Display, give it to the client.
+    // TODO: add support for version query.
+    if let Some(event) = events.pop() {
+        let display: DisplayVersionUpdatedEvent = bcs::from_bytes(&event.bcs[..])
+            .map_err(|e| anyhow!("Failed to deserialize 'VersionUpdatedEvent': {e}"))?;
+
+        Ok(Some(display))
     } else {
-        Err(anyhow!("Display object {display_object_id} does not exist"))?
+        Err(anyhow!("Display object for {object_type} does not exist"))?
```
Comment (666lcz) [inline crates/sui-json-rpc/src/read_api.rs:949]: I think the correct logic is to return `Ok(None)` if an event cannot be found. https://github.com/MystenLabs/sui/pull/9729

**Review by 666lcz — APPROVED** [review]

File: crates/sui-json-rpc/src/read_api.rs (hunk 1 above)
Comment (damirka) [reply crates/sui-json-rpc/src/read_api.rs:934]: Nope.

File: crates/sui-json-rpc/src/read_api.rs (hunk 2 above)
Comment (damirka) [reply crates/sui-json-rpc/src/read_api.rs:949]: Kk

**Pull request merged**